### PR TITLE
host: Move recent files to code mode

### PR DESCRIPTION
**Pull request by backspace**
This moves some code into a new `RecentFilesService` and changes more tests to use mock storage. The listing remains bare bones, there are followup issues to add more readable names and file extensions.

You can see it in action [here](https://hostrecent-code-mode-cs-5904.boxel-host-preview.stack.cards/?operatorModeEnabled=true&operatorModeState=%7B%22codePath%22%3A%22https%3A%2F%2Frealms-staging.stack.cards%2Fdrafts%2Fblog-post.gts%22%2C%22fileView%22%3A%22browser%22%2C%22openDirs%22%3A%5B%5D%2C%22stacks%22%3A%5B%5B%7B%22format%22%3A%22isolated%22%2C%22id%22%3A%22https%3A%2F%2Frealms-staging.stack.cards%2Fdrafts%2Findex%22%7D%5D%5D%2C%22submode%22%3A%22code%22%7D) though you’ll have to click around in the file browser before any entries will show.

### CODE REVIEW

File: packages/host/tests/acceptance/operator-mode-test.ts [hunk 1]
```diff
@@ -1542,7 +1545,9 @@ module('Acceptance | operator mode tests', function (hooks) {
 
       assert.dom('[data-test-search-sheet]').hasClass('results'); // Search open
 
-      await waitFor(`[data-test-search-result="${testRealmURL}Pet/mango"]`);
+      await waitFor(`[data-test-search-result="${testRealmURL}Pet/mango"]`, {
+        timeout: 2000,
```
Comment (backspace) [inline packages/host/tests/acceptance/operator-mode-test.ts:1448]: I finally added this after it failed [four times](https://github.com/cardstack/boxel/actions/runs/6153655328/job/16698511404).

**Comment by jurgenwerk** [comment]
The behaviour is unexpected to me. When I click on the first item it changes to something else

https://github.com/cardstack/boxel/assets/273660/5a228a40-297d-418b-a930-d3ac56a851eb

Also, when I click on any of the items, the list reoders which also feels awkward. Perhaps the list should stay the same when clicking on an item, and the list should change only when a new item is added on top of it.

**Comment by backspace** [comment]
hmm well it was specified in the issue:

> The list should exclude the currently open file

This PR is more about moving the list into code mode, it was already acting like this in `/code`. But we can have another issue if the behaviour should change.

**Review by habdelra — APPROVED** [review]
looks good

**Comment by backspace** [comment]
I’ve had persistent problems with the new test in this PR, it works locally but not in CI. I’ve made it `todo` for now and will open an issue to restore it.

**Pull request merged**